packages/devtools/lib/src/timeline/cpu_profile_model.dart

```diff
@@ -237,3 +237,15 @@ class CpuStackFrame {
     return buf.toString();
   }
 }
+
+int _stackFrameIdCompare(String a, String b) {
+  // Stack frame ids are structured as "140225212960768-24". We need to compare
+  // the number after the dash to maintain the correct order.
+  const dash = '-';
+  final aDashIndex = a.indexOf(dash);
+  final bDashIndex = b.indexOf(dash);
+  assert(aDashIndex != -1 && bDashIndex != -1);
```

Comment:
this will throw if the input has issues; do we want to do that, or fail w/o an exception (a string compare? some other method?)

Reply:
Good question. String compare will give us the wrong stack frame order for the profile, so I would say we should throw and show an error rather than display an inaccurate profile.

Reply:
`'Error reading performance profile'` or similar if we have trouble parsing the cpu profile data? sgtm if you think that's how we should handle it.